db/migrate/20170326222205_add_time_spent_to_library_entry.rb

```diff
@@ -0,0 +1,23 @@
+require 'update_in_batches'
+
+class AddTimeSpentToLibraryEntry < ActiveRecord::Migration
+  using UpdateInBatches
+
+  self.disable_ddl_transaction!
+
+  def up
+    add_column :library_entries, :time_spent, :integer
+
+    say_with_time 'Filling time_spent column' do
+      LibraryEntry.where(media_type: 'Anime').joins(:anime)
+        .update_in_batches('time_spent = anime.episode_length * progress')
```

Comment:
I'm pretty sure it'll just replace it with 0 (since this is Postgres math, not Ruby math), that's what it did last night when I was pulling numbers for josh 🤔 But it's possible that order matters, I did `progress * episode_length`

If you wanna be explicit you could use `coalesce(anime.episode_length, 0)`